File: SpaClient/src/interfaces/action-result.ts
import { ActionResultType } from '@/enums/action-result.type';
import { Direction } from '@/enums/direction.type';
import Point from './point';

export default interface ActionResult {
   resultType: ActionResultType;
   moveStart: Point;
   moveEnd: Point;
   directionStart: Direction;
   directionEnd: Direction;
}

File: Domain/Models/ActionResult.cs
using Domain.Enums;

namespace Domain.Models
{
    public class ActionResult
    {
        public ActionResultType ResultType { get; set; }
        public Point MoveStart { get; set; }
        public Point MoveEnd { get; set; }
        public DirectionType DirectionStart { get; set; }
        public DirectionType DirectionEnd{ get; set; }

        public ActionResult(WalkingSprite startState, WalkingSprite endState)
        {
            MoveStart = startState.Point;
            DirectionStart = startState.Direction;
            MoveEnd = endState.Point;
            DirectionEnd = startState.Direction;
        }
    }
}
